examples/with-apollo-and-redux/lib/withData.js

```diff
@@ -1,56 +1,77 @@
-import 'isomorphic-fetch'
 import React from 'react'
+import PropTypes from 'prop-types'
 import { ApolloProvider, getDataFromTree } from 'react-apollo'
-import { initClient } from './initClient'
-import { initStore } from './initStore'
+import initApollo from './initApollo'
+import initRedux from './initRedux'
 
-export default (Component) => (
-  class extends React.Component {
+export default ComposedComponent => {
+  class WithData extends React.Component {
     static async getInitialProps (ctx) {
-      const headers = ctx.req ? ctx.req.headers : {}
-      const client = initClient(headers)
-      const store = initStore(client, client.initialState)
+      let serverState = {}
 
-      const props = {
-        url: { query: ctx.query, pathname: ctx.pathname },
-        ...await (Component.getInitialProps ? Component.getInitialProps(ctx) : {})
+      // Evaluate the composed component's getInitialProps()
+      let composedInitialProps = {}
+      if (ComposedComponent.getInitialProps) {
```

Comment:
We probably shouldn't be using undocumented Next.js internals because when this code is actually implemented it'll be outside of the Next.js package.